pulpcore/cli/ansible/content.py

```diff
@@ -130,22 +175,35 @@ def content(ctx: click.Context, pulp_ctx: PulpContext, content_type: str) -> Non
     allowed_with_contexts=role_context,
     required=True,
 )
+@pulp_option(
+    "--collection",
+    help=_("Collection for this {entity}"),
+    allowed_with_contexts=signature_context,
+    required=True,
+)
 @pass_entity_context
 @pass_pulp_context
 def upload(
     pulp_ctx: PulpContext,
-    content_ctx: Union[PulpAnsibleRoleContext, PulpAnsibleCollectionVersionContext],
+    content_ctx: Union[
+        PulpAnsibleRoleContext,
+        PulpAnsibleCollectionVersionContext,
+        PulpAnsibleCollectionVersionSignatureContext,
+    ],
     file: IO[bytes],
-    chunk_size: int,
     **kwargs: Any,
 ) -> None:
 
     if isinstance(content_ctx, PulpAnsibleRoleContext):
-        artifact_href = PulpArtifactContext(pulp_ctx).upload(file, chunk_size)
+        artifact_href = PulpArtifactContext(pulp_ctx).upload(file)
         body = {"artifact": artifact_href}
         body.update(kwargs)
         result = content_ctx.create(body=body)
         pulp_ctx.output_result(result)
+    elif isinstance(content_ctx, PulpAnsibleCollectionVersionSignatureContext):
+        body = {"signed_collection": kwargs.get("collection")}
+        uploads = {"file": file.read()}
+        pulp_ctx.output_result(content_ctx.create(body=body, uploads=uploads))
     else:
         result = content_ctx.upload(file=file)
```

Comment:
Now this is the case for `PulpAnsibleCollectionVersion`, maybe make that more explicit with elif and raise an NotImplemented in the final else.
Also i think, collections can get arbitrarily large. We should support chunked upload (probably in a separate PR).

Reply:
+1 for being more explicit, and yes for supporting chunked upload. Going to need a new option on the upload endpoint in pulp_ansible as currently it only supports file uploads.